docs/development/pigs/pig-012.rst

```diff
@@ -0,0 +1,334 @@
+This module will contain also the set of different command line tools provided, where some small cleaning
+and refactoring may be needed (i.e. remove ``gammapy image`` command line tool)
+
+**Serialisation**
+
+The user could choose in the configuration file to serialise the products delivered by the tools 
+with the help of a boolean parameter. The serialisation would then be performed as follows:
+
+- Every tool in Gammapy knows how to serialise/read themselves
+- Composite objects will be serialised as a series of all of their components
+- Serialisation will be a mix of YAML (models, state) and FITS files (maps)
+- There will not be framework supporting different serialisation backends, we will have one way.
+
+**Session workflow**
+
+.. code::
+
+        $ mkdir analysis
+        $ cd analysis
+        $ edit gammapy_analysis_config.yaml
+        
+Then the user would type ``ipython``, ``juypter notebook`` or write a script with the code below.
+
+.. code::
+
+        from gammapy.analysis import Analysis
+
+        analysis = Analysis(config)
+
+        analysis.select_observations()
+        # Select observations using the parameters defined in the configuration file.
+
+        analysis.reduce_data()  # often slow, can be hours
+        # If the user wants they can save all results from data reduction and re-start later.
+        # This stores config, datasets, ... all the analysis class state.
+        # analysis.write()
+        # analysis = Analysis.read()
+
+        analysis.optimise()  # often slow, can be hours
+        # Again, we could write and read, do the slow things only once.
+        # e.g. supervisor comes in and asks about significance of some model component or whatever.
+        # analysis.write()
+        # analysis = Analysis.read()
+
+        # Since anything is accessible from the Analysis object
+        # many advanced use cases can be done with the Analysis API.
+        analysis.model("source_42").spectrum.plot()
+
+        # Should we need energy_binning for the SED points in config or only here?
+        sed = analysis.spectral_points("source_42", energy_binning)
+
+**Command line tools**
+
+In addition to ``gammapy analysis config``, we will have a ``gammapy analysis data_reduction``
+and a ``gammapy analysis optimise`` which perform the long processing tasks from the terminal 
+outside an ipython session or jupyter notebook, using all the information from the config 
+file and/or saved state.
+
+- ``gammapy analysis config``: dumps a template configuration file
+- ``gammapy analysis data_reduction``: performs a data reduction process
+- ``gammapy analysis optimise``: performs a model fitting process
+ 
+ 
+Outlook
+=======
+
+Some of the use cases not covered by this high-level interface API are the following:
+
+- Generation of simulated events and/or counts
+- Iterative source detection methods
+- Lightcurves
+
```

Comment:
Adding LC is quite important. If the manpower allows it, one should do it as soon as possible

Reply:
Yes, LC is important. But note that there is a separate PIG for LC: #1451 
So if we put more info on LC here, it starts to get controversial and we make the PIG process inefficient because we discuss in two places.
So suggest to keep things here as-is (except to reference #1451 if it doesn't exist already), and then to help finish #1451 ASAP.

Reply:
Yes, I've rephrased the item and added a link to the PIG on Light Curves in the paragraph below.